aiidalab_widgets_base/computational_resources.py

```diff
@@ -225,12 +245,56 @@ def _setup_new_code(self, _=None):
                 }
 
 
+class SshConnectionState(enum.Enum):
+    waiting_for_input = -1
+    enter_password = 0
+    success = 1
+    keys_already_present = 2
+    do_you_want_to_continue = 3
+    no_keys = 4
+    unknown_hostname = 5
+    connection_refused = 6
+    end_of_file = 7
```

Comment:
We can probably use [`auto`](https://docs.python.org/3.7/library/enum.html#using-automatic-values) here, right?

Reply:
As we discussed offline, this needs to correspond to the index of `SSH_POSSIBLE_RESPONSES`.